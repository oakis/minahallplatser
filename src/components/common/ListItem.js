import React from 'react';
import { TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { component } from '../style';
import { Text } from '../common';

export const ListItem = ({ text, icon = null, pressItem, pressIcon = null, iconVisible = false, iconColor = '#000' }) => {
	return (
		<TouchableOpacity
			onPress={pressItem}
			style={component.listitem.view}
		>
			<Text style={component.listitem.text}>
				{text}
			</Text>
			{(!iconVisible) ? null :
				<Icon
					style={[component.listitem.icon, { color: iconColor }]}
					name={icon}
					size={24}
					onPress={pressIcon}
				/>
			}
		</TouchableOpacity>
	);
};
