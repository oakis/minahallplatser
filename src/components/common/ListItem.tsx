import React from 'react';
import {TouchableOpacity, Image, View} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {component} from '@style';
import {Text} from '@common';

type ListItemProps = {
  text: string;
  icon: string | null;
  pressItem: () => void;
  pressIcon: () => void;
  iconVisible: boolean;
  iconColor: string;
  style: object;
  avatar: string | null;
};

export const ListItem = ({
  text,
  icon = null,
  pressItem,
  pressIcon = () => {},
  iconVisible = false,
  iconColor = '#000',
  style = {},
  avatar = null,
}: ListItemProps): JSX.Element => {
  return (
    <TouchableOpacity
      onPress={pressItem}
      style={[component.listitem.view, style]}>
      <Text style={component.listitem.text}>{text}</Text>
      <View style={component.listitem.icon}>
        {!iconVisible || avatar !== null ? null : (
          <Icon
            style={{color: iconColor}}
            name={icon}
            size={24}
            onPress={pressIcon}
          />
        )}
        {avatar === null ? null : (
          <Image
            style={{width: 24, height: 24, borderRadius: 12}}
            source={{uri: avatar}}
          />
        )}
      </View>
    </TouchableOpacity>
  );
};
