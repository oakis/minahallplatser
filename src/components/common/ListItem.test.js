import React from 'react';
import { shallow } from 'enzyme';
import { ListItem } from './ListItem';

it('should match snapshot', () => {
    const wrapper = shallow(<ListItem />);
    expect(wrapper).toMatchSnapshot();
});
