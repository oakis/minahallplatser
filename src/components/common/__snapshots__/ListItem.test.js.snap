// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<TouchableOpacity
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
      },
      null,
    ]
  }
>
  <DefaultFont
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 18,
        "justifyContent": "flex-start",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 24,
      }
    }
  />
</TouchableOpacity>
`;
